core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java

```diff
@@ -321,19 +296,19 @@ private static SQLAdmin createAdminApiClient(HttpRequestInitializer requestIniti
 
   private static class ApplicationDefaultCredentialFactory implements CredentialFactory {
     @Override
-    public Credential create() {
-      GoogleCredential credential;
+    public HttpRequestInitializer create() {
+      GoogleCredentials credentials;
       try {
-        credential = GoogleCredential.getApplicationDefault();
+        credentials = GoogleCredentials.getApplicationDefault();
       } catch (IOException err) {
         throw new RuntimeException(
             "Unable to obtain credentials to communicate with the Cloud SQL API", err);
       }
-      if (credential.createScopedRequired()) {
-        credential =
-            credential.createScoped(Collections.singletonList(SQLAdminScopes.SQLSERVICE_ADMIN));
+      if (credentials.createScopedRequired()) {
+        credentials =
+            credentials.createScoped(Collections.singletonList(SQLAdminScopes.SQLSERVICE_ADMIN));
       }
-      return credential;
+      return new HttpCredentialsAdapter(credentials);
```

Comment:
So why not make this method return `com.google.auth.Credentials` instead and do the wrapping in the caller?
You would change the method signature to `Credentials create()` as well.

Reply:
That would be a source code breaking change, `com.google.api.client.auth.oauth2.Credential` implements `com.google.api.client.http.HttpRequestInitializer` so this way the PR is source code compatible - existing implementations of `com.google.cloud.sql.CredentialFactory` will not need any source code changes, only be recompiled against the new library version. `com.google.auth.Credentials` on the other hand is a completely separate API and would require all existing implementations to change.

Reply:
You right. Thanks for the clarification.